lib/shopping_cart.rb

```diff
@@ -0,0 +1,60 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+  attr_accessor :products_in_cart
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+    @products_in_cart = 0
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details = {
+      name: "King Soopers", 
+      capacity: 30
+    }
+  end
+
+  def total_number_of_products
+    @products_in_cart = 0
+    @products.each do |product|
```

Comment:
This works and is functional which is a great start, but like I mentioned above we don't need and instance variable here, this can just be a local variable called `products_in_cart = 0`

Also there is an enumerable called `sum` we can use here that add integer values for us, so instead of `#each` we can have the following code.


```
  @products.sum do |product|
    product.quantity
  end
```